docs/set-discovery-password.md

```diff
@@ -1,81 +1,74 @@
+To allow that, the user must have a password set. In order to set a password,
+it's possible to use the following script which allows editing the discovery
+ISO's ignition file core user with a password
 
-For more information about the API and its various authentication methods, see [this document](cloud.md).
+# Change ISO `core` user password script
 
-# Modify password script
-```sh
+The following script takes the path of the downloaded discovery ISO file as its
+first argument, prompts the user to enter a password, and creates a similarly
+named ISO file in the same folder as the original one but with password login
+enabled.
+
+```bash
 #!/bin/bash
 
 set -euo pipefail
 
-if ! ocm token 2>/dev/null >/dev/null; then
-    echo "Failed to run 'ocm token' command, please see the assisted_service/docs/cloud.md doc for authentication information"
+if [[ $# -ne 1 ]]; then
+    echo "Usage: $0 <path to discovery .iso>"
     exit 1
 fi
 
-## User specific configuration <-----------
-WANTED_PASSWORD=mypass
-TOKEN=$(ocm token)
-OCM_API_ENDPOINT="https://api.openshift.com/api/" 
-if true; then
-    echo "Don't forget to modify the script with your cluster ID, delete this warning after you did"
+if [[ ! -f $1 ]]; then
+    echo "ERROR: Discovery ISO not found at $1"
     exit 1
 fi
-INFRA_ENV_ID="243e09fb-d924-42a1-bad9-b78638b767d9" # Get from Assisted Installer API
-DEST_ISO_FILE="changed_password.iso"
-###############################
 
-function log() {
-    if [[ ! $? == 0 ]]; then
-        echo "Script enountered an error"
-        exit 1
-    fi
+DISCOVERY_ISO_HOST_PATH="$1"
+DISCOVERY_ISO_HOST_DIR=$(dirname "$DISCOVERY_ISO_HOST_PATH")
+function COREOS_INSTALLER() {
+    podman run -v "$DISCOVERY_ISO_HOST_DIR":/data --rm quay.io/coreos/coreos-installer:release "$@"
 }
 
-trap log EXIT
-
-echo "Password set log" > .set_iso_password_log
+ISO_NAME=$(basename "$DISCOVERY_ISO_HOST_PATH" .iso)
 
-DISCOVERY_IGN_URL=$OCM_API_ENDPOINT/assisted-install/v2/infra-envs/$INFRA_ENV_ID/downloads/'files?file_name=discovery.ign'
-
-if ! curl --fail -s ${DISCOVERY_IGN_URL} -H "Authorization: Bearer $TOKEN" >/dev/null; then
-    echo "Can't seem to find a discovery.ign, please generate a discovery ISO file using the UI first"
-    exit 1
-fi
+# Container paths
+DISCOVERY_ISO_PATH=/data/${ISO_NAME}.iso
+DISCOVERY_ISO_WITH_PASSWORD=/data/${ISO_NAME}_with_password.iso
 
-echo Downloading the original ignition file into the ORIGINAL_IGNITION variable
-ORIGINAL_IGNITION=$(curl --fail -s ${DISCOVERY_IGN_URL} -H "Authorization: Bearer $TOKEN")
+# Host output path
+DISCOVERY_ISO_WITH_PASSWORD_HOST=$(dirname "$DISCOVERY_ISO_HOST_PATH")/$(basename "$DISCOVERY_ISO_WITH_PASSWORD")
 
-echo === Original ignition === >> .set_iso_password_log
-echo "$ORIGINAL_IGNITION" >> .set_iso_password_log
-echo ========================= >> .set_iso_password_log
-echo >> .set_iso_password_log
+# Prompt
+USER_PASSWORD=$(mkpasswd --method=SHA-512)
```

Comment:
mkpasswd is an ambiguous tool. It can either be the mkpasswd tool you're using here, or part of expect (different tool with different arguments):
```
$ rpm -qf $(which mkpasswd)
expect-5.45.4-5.el8.x86_64
$ mkpasswd --method=SHA-512
passwd: bad argument --method=SHA-512: unknown option
```

I suggest using openssl to generate the user password:
```
read -sp 'Password: ' pw
echo ''
USER_PASSWORD=$(openssl passwd -6 --stdin <<< $pw)
unset pw
```

Reply:
Good catch, thank you

Reply:
That works